VenafiTppPS/Code/Public/Get-TppCertificate.ps1

```diff
@@ -121,19 +151,44 @@ function Get-TppCertificate {
 
         $params.Body.CertificateDN = $Path
 
-        if ( $PSBoundParameters.ContainsKey('IncludePrivateKey') ) {
+        if ($IncludePrivateKey.IsPresent) {
 
             # validate format to be able to export the private key
             if ( $Format -in @("Base64", "DER", "PKCS #7") ) {
                 Write-Error "Format '$Format' does not support private keys"
                 Return
             }
 
-            $params.Body.Add('IncludePrivateKey', $IncludePrivateKey)
+            $params.Body.Add('IncludePrivateKey', $true)
             $plainTextPassword = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($SecurePassword))
             $params.Body.Add('Password', $plainTextPassword)
         }
 
+        if ($Format -in @("Base64 (PKCS #8)", "DER", "PKCS #7")) {
+            if (-not ([string]::IsNullOrEmpty($FriendlyName))) {
+                Write-Error "Only Base64, JKS, PKCS #12 formats support FriendlyName parameter"
+                Return
+            }
+        }
+        else {
+            if ($Format -ieq 'JKS' -and [string]::IsNullOrEmpty($FriendlyName)) {
+                Write-Error "JKS format requires FriendlyName parameter to be set"
+                Return
+            }
+        }
+
+        if (-not [string]::IsNullOrEmpty($FriendlyName)) {
+            $params.Body.Add('FriendlyName', $FriendlyName)
+        }
+
+        if (($Format -in @("Base64 (PKCS #8)", "DER")) -and $IncludeChain.IsPresent)
+        {
+            Write-Error "IncludeChain is only supported when Format is Base64, JKS, PKCS #7, or PKCS #12"
+            Return
+        }
+
+        $params.Body.Add('IncludeChain', $IncludeChain.IsPresent)
```

Comment:
please remove `.IsPresent`

Reply:
Hi @gdbarron, do you mean replace this line

```powershell
@params.Body.Add('IncludeChain', $IncludeChain.IsPresent)
```

with

```powershell
if ($PSBoundParameters.ContainsKey('IncludeChain')
{
     @params.Body.Add('IncludeChain', $true)
}
```